packages/runtime/src/next/components/head-tags.tsx

```diff
@@ -0,0 +1,66 @@
+import { ComponentPropsWithoutRef } from 'react'
+import { useIsPagesRouter } from '../../components/hooks/useIsPagesRouter'
+import Head from 'next/head'
+
+export function PageTitle({ children, ...props }: ComponentPropsWithoutRef<'title'>) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <title {...props}>{children}</title>
+      </Head>
+    )
+  }
+  return <title {...props}>{children}</title>
+}
+
+export function PageMeta(props: ComponentPropsWithoutRef<'meta'>) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <meta {...props} />
+      </Head>
+    )
+  }
+  return <meta {...props} />
+}
+
+// https://react.dev/reference/react-dom/components/link#props
+type ReactLinkProps = {
+  precedence?: 'reset' | 'low' | 'medium' | 'high'
+}
+type PageLinkProps = ComponentPropsWithoutRef<'link'> & ReactLinkProps
+
+export function PageLink({ precedence, ...props }: PageLinkProps) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <link {...props} />
+      </Head>
+    )
+  }
+  //@ts-expect-error
```

Comment:
nit: we should add a space between `//` and `@`.

Also, can we add a note of what the expected error is so that this is more "deletable"? For example:

```ts
// @ts-expect-error: the current version of `@types/react-dom` doesn't include types for `precedence` since it's a React Canary feature. See https://react.dev/reference/react-dom/components/link. The functionality is still supported because Next.js App Router uses a React Canary.
```

That's a pretty verbose comment and perhaps we can do something shorter but it does specify why the error happens and also why we're ok with it happening (i.e., Next.js does support it).